auslib/blobs/gmp.py

```diff
@@ -61,6 +61,15 @@ def getInnerXML(self, updateQuery, update_type, whitelistedDomains, specialForce
 
         return vendorXML
 
+    def validate(self, *args, **kwargs):
+        Blob.validate(self, *args, **kwargs)
+        hashes = dict(SHA1=40, SHA256=64, SHA512=128)
```

Comment:
I think I wasn't quite clear on what I meant here...I think that the part that validates the hash length can be factored out into its own function, eg:
def validate_hash_length(type_, hash):

...and then that can be called from classes like ReleaseBlobBase and GMPBlobV1.

Does that make sense?